Month2/week1/Month2-Project Week 1-Remembering the Bases/VideoGame/Steam.cpp

```diff
@@ -0,0 +1,408 @@
+#include "Steam.h"
+
+FSteam::FSteam() 
+{
+	ActiveCommand = EMenuCommand::MainMenu;
+}
+
+void FSteam::RunApp() 
+{
+	while (ActiveCommand != EMenuCommand::Exit) 
+	{
+		switch (ActiveCommand)
+		{
+		case FSteam::EMenuCommand::MainMenu:
+			OpenMainMenu();
+			break;
+		case FSteam::EMenuCommand::AddGames:
+			OpenAddgamesMenu();
+			break;
+		case FSteam::EMenuCommand::Categories:
+			OpenCategoriesMenu();
+			//Actions();
+			break;
+
+		case FSteam::EMenuCommand::DisplayGames:
+			OpenVisualiceGamesMenu();
+			break;
+		default:
+			break;
+		}
+	}
+}
+/*void FSteam::Actions()
+{
+	switch (ActionChoice)
+	{
+	case FSteam::Action::CreateCategory:
+		CreateCategory();
+		break;
+	case FSteam::Action::DeleteCategory:
+		DeleteCategory();
+		break;
+	default:
+		break;
+	}
+
+}*/
+void FSteam::OpenMainMenu()
+{
+	system("CLS");
+
+	std::cout << "Welcome to Steam. What do you want to do?" << std::endl;
+	std::cout << "1.Add Games" << std::endl;
+	std::cout << "2.Browse Category" << std::endl;
+	std::cout << "3.Visualice Games" << std::endl;
+	std::cout << "4.Exit" << std::endl;
+	int OptionChoice;
+	std::cin >> OptionChoice;
+
+	if (!std::cin.good() || OptionChoice < 1 || OptionChoice > 4) 
+	{
+		std::cout << "Invalid Input, enter your new choice" << std::endl;
+		std::cin.clear();
+		std::cin.ignore(1000, '\n');
+		std::cin >> OptionChoice;
+	}
+	ActiveCommand = (EMenuCommand)OptionChoice;
+	system("CLS");
+}
+
+void FSteam::OpenAddgamesMenu()
+{
+	std::cin.clear();
+	std::cin.ignore(1000, '\n');
+	system("CLS");
+	std::cout << "Adding a VideoGame �Fantastic!" << std::endl;
+	int Choice = 1;
+	do 
+	{
+		system("CLS");
+		std::cout << "Please enter the name of the Game" << std::endl;
+		std::string EntryGameName;
+		std::getline(std::cin, EntryGameName);
+		while (!std::cin.good())
+		{
+			std::cin.clear();
+			std::cin.ignore(1000, '\n');
+			std::cout << "Invalid input, please enter your Game name again:" << std::endl;
+			std::getline(std::cin, EntryGameName);
+		}
+
+		std::cout << "Please enter the name of the Studio " << std::endl;
+		std::string EntryStudioName;
+		std::getline(std::cin, EntryStudioName);
+		while (!std::cin.good() || EntryGameName=="")
```

Comment:
This validation code is repeated many many many times. This is when a function should be considered. You want to avoid code duplication as much as possible (which should be ideally always).

Reply:
This is also an incredibly long method. This would be simplified a whole lot if you had a function for validation.